src/components/VmStatus/VmStatus.js

```diff
@@ -126,74 +102,137 @@ VmStatuses.propTypes = {
 
 export const VmStatus = ({ vm, pods, migrations, verbose }) => {
   const statusDetail = getVmStatus(vm, pods, migrations);
-  switch (statusDetail.status) {
-    case VM_STATUS_V2V_CONVERSION_IN_PROGRESS:
-      return (
-        <StateV2VConversionInProgress {...statusDetail} linkTo={getSubPagePath(statusDetail.pod, PodModel, 'events')} />
-      );
-    case VM_STATUS_V2V_CONVERSION_PENDING:
-      return (
-        <StateV2VConversionPending {...statusDetail} linkTo={getSubPagePath(statusDetail.pod, PodModel, 'events')} />
-      );
-    case VM_STATUS_V2V_CONVERSION_ERROR:
-      return (
-        <StateV2VConversionError {...statusDetail} linkTo={getSubPagePath(statusDetail.pod, PodModel, 'events')} />
-      );
-    case VM_STATUS_IMPORTING:
-      return (
-        <StateImporting
-          linkTo={getSubPagePath(statusDetail.pod, PodModel, 'events')}
-          additionalText={verbose ? getAdditionalImportText(statusDetail.pod) : ''}
-        />
-      );
-    case VM_STATUS_IMPORT_ERROR:
-      return (
-        <StateError linkTo={getSubPagePath(statusDetail.pod, PodModel, 'events')} message={statusDetail.message}>
-          Importer Error
-          {verbose ? getAdditionalImportText(statusDetail.pod) : ''}
-        </StateError>
-      );
-    case VM_STATUS_MIGRATING:
-      return <StateMigrating />; // TODO: add linkTo once migration monitoring page is available
-    case VM_STATUS_OFF:
-      return <StateOff />;
-    case VM_STATUS_RUNNING:
-      return <StateRunning linkTo={getSubPagePath(statusDetail.launcherPod, PodModel)} />;
-    case VM_STATUS_VMI_WAITING:
-      return <StateVmiWaiting linkTo={getSubPagePath(vm, VirtualMachineModel, 'events')} />;
-    case VM_STATUS_STARTING:
-      return (
-        <StateStarting
-          linkTo={getSubPagePath(statusDetail.launcherPod, PodModel, 'events')}
-          message={statusDetail.message}
-        />
-      );
-    case VM_STATUS_POD_ERROR:
-      return (
-        <StateError
-          linkTo={getSubPagePath(statusDetail.launcherPod, PodModel, 'events')}
-          message={statusDetail.message}
-        >
-          Pod Error
-        </StateError>
-      );
-    case VM_STATUS_ERROR:
-      return (
-        <StateError linkTo={getSubPagePath(vm, VirtualMachineModel, 'events')} message={statusDetail.message}>
-          VM Error
-        </StateError>
-      );
-    default:
-      return <StateUnknown />; // Let's hope this state is tentative and will fit former conditions soon
-  }
-};
+  const additionalText =
+    verbose && statusDetail.pod && statusDetail.pod.metadata.labels ? getAdditionalImportText(statusDetail.pod) : null;
+
+  const statusProps = {
```

Comment:
IMO I think it would be better to not use the wrapper and create ` <PopoverStatus ... >` or `<Status ...>` accordingly here (we do not need special components like StateV2VConversionInProgress). 

But the approach with the wrapper is a bit confusing (just a lot of `if`s) and doesn't bring much of  additional functionality - so I think it is easier to just declare the components here.

Reply:
plus it is a bit inefficient to create so much markup in each render

Reply:
the ifs could be also put inside the components - if progress == null - do not render the component

Reply:
thoughts on all of this?

Reply:
> thoughts on all of this?

Nice, it is a lighter implementation :dark_sunglasses: , reverting back to switch :smile: